src/main/java/io/khasang/freefly/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,38 @@
+package io.khasang.freefly.dao.impl;
+
+import io.khasang.freefly.dao.OrderDao;
+import io.khasang.freefly.entity.Order;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class OrderDaoImpl extends BasicDaoImpl<Order> implements OrderDao {
+
+    @Autowired
+    SessionFactory sessionFactory;
+
+    public OrderDaoImpl(Class<Order> entityClass) {
+        super(entityClass);
+    }
+
+    @Override
+    public Order update(Order entity) {
+        getSessionFactory().update(entity);
+        return entity;
+    }
+
+    private Session getSessionFactory() {
+        return sessionFactory.getCurrentSession();
+    }
+
+    @Override
+    public Order deleteById(long id) {
+        Order entity = getById(id);
```

Comment:
@Override
    public Cat deleteCat(long id) {
        if (Objects.nonNull(getCatById(id))) {
            return catDao.delete(getCatById(id));
        }

        throw new IllegalArgumentException("cat with id: " + id + " not exist");
    }